Break down Move doc into sub-pages (#3319)
MystenLabs/sui: doc/src/build/move/sui-move-library.md

To transfer an object `obj` (whose owner is an account address) to another object `owner`:
```
transfer::transfer_to_object(obj, &mut owner);
```
This function returns a `ChildRef` instance that cannot be dropped arbitrarily. It can be stored in the parent as a field.
Sometimes we need to set the child field of a parent while constructing it. In this case, we don't yet have a parent object to transfer into. In this case, we can call the `transfer_to_object_id` API. Example:
```
let parent_info = object::new(ctx);
let child = Child { info: object::new(ctx) };
let (parent_id, child_ref) = transfer::transfer_to_object_id(child, parent_info);
let parent = Parent {
    info: parent_info,
    child: child_ref,
};
transfer::transfer(parent, tx_context::sender(ctx));
```
To transfer an object `child` from one parent object to a new parent object `new_parent`, we can use the following API:
```
transfer::transfer_child_to_object(child, child_ref, &mut new_parent);
```
Note that in this call, we must also have the `child_ref` to prove the original ownership. The call will return a new instance of `ChildRef` that the new parent can maintain.

> To transfer an object `child` from an object to an account address `recipient`, we can use the following API:

Reviewer: this is more a transfer method/operation of the API than the API itself?

```
transfer::transfer_child_to_address(child, child_ref, recipient);
```

> This call also requires to have the `child_ref` as proof of original ownership.

Reviewer: ```suggestion
The original `child_ref` must be included in this call as proof of original ownership.
```

After this transfer, the object will be owned by `recipient`.

> More examples of how objects can be transferred and owned can be found in

Reviewer: ```suggestion
You can view additional examples of how objects can be transferred and owned in
```

[object_owner.move](https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/unit_tests/data/object_owner/sources/object_owner.move).

### Freeze an object

> To make an object `obj` shared and immutable, one can call:

Reviewer: ```suggestion
The following call demonstrates how to make an object `obj` shared and immutable:
```

```
transfer::freeze_object(obj);
```

> After this call, `obj` becomes immutable which means it can never be mutated or deleted. This process is also irreversible: once an object is frozen, it will stay frozen forever. An immutable object can be used as reference by anyone in their Move call.

Reviewer: ```suggestion
After this call, `obj` becomes immutable, which means it can never be mutated or deleted. This process is also irreversible: once an object is frozen, it will stay frozen forever. An immutable object can be used as reference by anyone in their Move call.
```

### Share an object (experimental)
This feature is still in development. It only works in Move for demo purpose, and doesn't yet work in Sui.

To make an object `obj` shared and mutable, one can call:
```
transfer::share_object(obj);
```

> After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be deleted, transferred or embedded in another object as a field.

Reviewer: shared by everyone -or- shared with everyone?

> After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be deleted, transferred or embedded in another object as a field.

Reviewer: ```suggestion
After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be deleted, transferred, or embedded in another object as a field.
```

> After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be deleted, transferred or embedded in another object as a field.

Reviewer: ```suggestion
After this call, `obj` stays mutable, but becomes shared by everyone, i.e. anyone can send a transaction to mutate this object. However, such an object cannot be deleted, transferred, or embedded in another object as a field.
```

> Shared mutable object can be powerful in that it will make programming a lot simpler in many cases. However shared object is also more expensive to use: it requires a full sequencer (a.k.a. a consensus engine) to order the transactions that touch the shared object, which means longer latency/lower throughput and higher gas cost. One can see the difference of the two programming schemes between not using shared object vs using shared object by looking at the two different implementations of TicTacToe: [No Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/tic_tac_toe.move) vs. [Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/shared_tic_tac_toe.move).

Reviewer: ```suggestion
A shared mutable object can be powerful in that it makes programming a lot simpler in many cases. However, a shared object is also more expensive to use. A shared object requires a full sequencer (a.k.a. a consensus engine) to order the transactions that touch the shared object. This results in longer latency/lower throughput and higher gas cost. One can see the difference of the two programming schemes between not using shared object vs using shared object by looking at the two different implementations of TicTacToe: [No Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/tic_tac_toe.move) vs. [Shared Object](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/games/sources/shared_tic_tac_toe.move).
```

## Transaction context

> `TxContext` module provides a few important APIs that operate based on the current transaction context.

Reviewer: important operations/methods of the API? I think the API is what I installed - Sui. This is a method of the API?

To create a new ID for a new object:
```
// assmue `ctx` has type `&mut TxContext`.
let info = sui::object::new(ctx);
```

To obtain the current transaction sender's account address:
```
sui::tx_context::sender(ctx)
```

## Next steps
Now that you are familiar with the Move language, as well as with how
to develop and test Move code, you are ready to start looking at and
playing with some larger
[examples](../../explore/examples.md) of Move

> programs, such as implementation of the tic-tac-toe game or a more

Reviewer: really long sentence
```suggestion
programs. The examples include implementation of the tic-tac-toe game, and a more
```

> fleshed out variant of a fantasy game similar to the one we have been

Reviewer: Fleshed out can be confusing for ESL and may not localize well
```suggestion
developed variant of a fantasy game similar to the one we have been
```
